src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClient.cs

```diff
@@ -200,8 +164,8 @@ public List<PSRoleDefinition> FilterRoleDefinitionsByCustom(string scope, bool s
         /// <returns>The created role assignment object</returns>
         public PSRoleAssignment CreateRoleAssignment(FilterRoleAssignmentsOptions parameters)
         {
-            Guid principalId = ActiveDirectoryClient.GetObjectId(parameters.ADObjectFilter);
-            Guid roleAssignmentId = RoleAssignmentNames.Count == 0 ? Guid.NewGuid() : RoleAssignmentNames.Dequeue();
+			string principalId = ActiveDirectoryClient.GetObjectId(parameters.ADObjectFilter).ToString();
+			string roleAssignmentId = (RoleAssignmentNames.Count == 0 ? Guid.NewGuid() : RoleAssignmentNames.Dequeue()).ToString();
```

Comment:
any other places in this PR

---
In reply to: [126038520](https://github.com/Azure/azure-powershell/pull/4257#discussion_r126038520) [](ancestors = 126038520)